.github/workflows/pull-request.yaml

```diff
@@ -0,0 +1,60 @@
+name: pull request
+on:
+  pull_request:
+  push:
+    branches: [main]
+jobs:
+  rust:
+    strategy:
+      matrix:
+        include:
+          - rust: 1.47.0
+            examples: false
+            continue-on-error: false
+          - rust: stable
+            examples: true
+            continue-on-error: false
+          - rust: beta
+            examples: false
+            continue-on-error: true
+          - rust: nightly
+            examples: false
+            continue-on-error: true
+    runs-on: ubuntu-latest
+    continue-on-error: ${{ matrix.continue-on-error }}
+    env:
+      PK: ${{ secrets.DOCKER_NAME }}
+      INFURA_PROJECT_ID: ${{ secrets.DOCKER_PASSWORD }}
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions-rs/toolchain@v1
+        with:
+          toolchain: ${{ matrix.rust }}
+          profile: minimal
+          components: rustfmt, clippy
+          default: true
+      - uses: Swatinem/rust-cache@v1
+      - run: cargo fmt -- --check
+      - run: cd examples/truffle && yarn --frozen-lockfile && yarn build
+      - run: cargo clippy --all-features --all-targets -- -D warnings
+      - run: cargo test --all-features
+      - run: |
```

Comment:
Is it possible to run this as a separate "job" so that it is run in parallel?

Reply:
It is probably possible but I'm not sure how to do this in a nice way. The whole job is so fast (2 or 3 minutes) anyway that it doesn't matter much. But agree it's something we can keep in mind for the future if we want to optimize it more.